docker/Dockerfile

```diff
@@ -229,19 +224,19 @@ RUN set -e; \
     truncate -s0 ~/.bash_history;
 
 RUN set -e; \
-    if [ -z $(which bazel) ]; then \
+    if ! which bazel >/dev/null ; then \
         BAZEL_INSTALL_DIR='/usr/local/bin/'; \
         BAZEL_VERSION='5.2.0'; \
         BAZEL_INSTALL_PATH="$BAZEL_INSTALL_DIR/bazel-$BAZEL_VERSION"; \
         BAZEL_ARCH=$(case "$TARGETARCH" in \
-                     'amd64') \
-                         echo 'x86_64'; \
-                         ;; \
-                     'arm64') \
-                         echo 'arm64'; \
-                         ;; \
-                   esac); \
-        sudo wget -O "$BAZEL_INSTALL_PATH" "https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-linux-$BAZEL_ARCH"; \
+        'amd64') \
+            echo 'x86_64'; \
+            ;; \
+        'arm64') \
+            echo 'arm64'; \
+            ;; \
+        esac); \
+        sudo curl -o "$BAZEL_INSTALL_PATH" "https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-linux-$BAZEL_ARCH"; \
```

Comment:
Add retry logic to all downloading: https://github.com/xkszltl/Roaster/blob/8c8af5f0052d3307bdef38cb4bfbb7e18621dd1a/pkgs/repo.sh#L71

Reply:
set retry opts for curl and wrap with until on top.

Reply:
`--retry` should be enough. If failure the return value is non-zero and the build will stop, which is a desired behavior.

Reply:
No, retry itself has some limit on the failure type it can retry, basically if handles some flaky connections but not higher/lower level errors. In rough network condition we don't want it to fail, we want it to try really hard by itself.

Reply:
Do you mean that we remove `--retry` but retry it manually by `until`? Then we need to download the signature first and then check the hash to figure out if it is correctly downloaded.... I thought it is a bit overkill.

Reply:
No, both, --retry won't retry all failures, and depends on version of curl, there're a few extra --retry options for covering more failures, but still not all. We don't use only `until` because the error type handled by --retry has a different distribution than other failures, and not all of then worth a process launching overhead. For example, some CDN may return 404 for the first few access during warm up, that's not a network error, and curl itself should not retry. But we have L7 knowledge saying that doesn't make sense and should keep trying.